src/components/Breadcrumb/Breadcrumb.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Breadcrumb as CarbonBreadcrumb,
+  OverflowMenu as CarbonOverflowMenu,
+} from 'carbon-components-react';
+import useResizeObserver from 'use-resize-observer';
+
+import BreadcrumbOverflowItem from './BreadcrumbOverflowItem';
+import './_breadcrumb.scss';
+
+const propTypes = {
+  /** Pass in the BreadcrumbItem's for your Breadcrumb */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  children: null,
+};
+
+const Breadcrumb = props => {
+  const { children, ...other } = props;
+  const [breadcrumbItemsWidth, setBreadcrumbItemsWidth] = useState([0]);
+  const [totalWidth, setTotalWidth] = useState(0);
+  const { ref: breadcrumbRef, width: containerWidth } = useResizeObserver({
+    useDefaults: false,
+  });
+  // const [breadcrumbChildren, setBreadcrumbChildren] = useState(children);
+  const [overflowIndex, setOverflowIndex] = useState(0);
+  const childRef = React.Children.map(children, useRef);
+
+  const childrenWithRef = React.Children.map(children, (child, index) =>
+    React.cloneElement(child, {
+      ref: childRef[index],
+      key: index,
+    })
+  );
+
+  /** intial to get children total width */
+  useEffect(() => {
+    // measure child total width
+    const itemsWidth = [];
+    // const breadcrumb = breadcrumbRef.current;
+    // const breadcrumItems = breadcrumb.querySelectorAll('li.bx--breadcrumb-item');
+
+    // if (breadcrumItems.length > 0) {
+    //   for (let i = 0; i <= breadcrumItems.length - 1; i += 1) {
+    //     itemsWidth.push(breadcrumItems[i].offsetWidth);
+    //   }
+    //   setBreadcrumbItemsWidth(itemsWidth);
+    //   const curTotalWidth = itemsWidth.reduce((acc, cur) => acc + cur);
+    //   setTotalWidth(curTotalWidth);
+    // }
+
+    // console.log("parent :", breadcrumbRef.current);
+    // console.log("children : ", childrenWithRef);
+    for (let i = 0; i < childrenWithRef.length; i += 1) {
+      // console.log("child with ref: ", childrenWithRef[i].ref.current);
+      // console.log("children height x width: ", childrenWithRef[i].ref.current.getBoundingClientRect());
+      itemsWidth.push(childrenWithRef[i].ref.current.getBoundingClientRect().width);
+      setBreadcrumbItemsWidth(itemsWidth);
+      const curTotalWidth = itemsWidth.reduce((acc, cur) => acc + cur);
+      setTotalWidth(curTotalWidth);
```

Comment:
I don't think we should set state on every iteration. It seems your goal here is to add up the total width of the children combined. That should be calculated and then `setTotalWidth` should be called _once_, after the children have been iterated through.